### ISSUE TITLE
[Bug][iOS] CollectionView does not appear if in a StackLayout AND Page is not wrapped in a NavigationPage

### ISSUE THREAD

**Issue by jgold6** [xamarin/Xamarin.Forms#6834]
### Description
On iOS, `CollectionView` does not appear if in a `StackLayout` AND the page is not wrapped in a `NavgationPage`. 

If the `VerticalOptions` for the `CollectionView` is set to `FillAndExpand` then the `CollectionView` appears as expected. 

Issue was initially reported here:
https://stackoverflow.com/questions/56719398/how-to-render-collectionview-inside-stacklayout?noredirect=1

### Steps to Reproduce

1. Create a new Blank Forms app (using VS for Mac on my tests)
2. Update Forms and Essentials packages.
3. Add the following to the `StackLayout` in `MainPage.xaml` after the `Label`:
```
        <CollectionView  BackgroundColor="Accent"
               ItemsLayout="HorizontalList"         
               >
            <CollectionView.ItemsSource>
                <x:Array Type="{x:Type x:String}">
                    <x:String>Baboon</x:String>
                    <x:String>Capuchin Monkey</x:String>
                    <x:String>Blue Monkey</x:String>
                    <x:String>Squirrel Monkey</x:String>
                    <x:String>Golden Lion Tamarin</x:String>
                    <x:String>Howler Monkey</x:String>
                    <x:String>Japanese Macaque</x:String>
                    <x:String>Mandrill</x:String>
                    <x:String>Proboscis Monkey</x:String>
                    <x:String>Red-shanked Douc</x:String>
                    <x:String>Gray-shanked Douc</x:String>
                    <x:String>Golden Snub-nosed Monkey</x:String>
                    <x:String>Black Snub-nosed Monkey</x:String>
                    <x:String>Tonkin Snub-nosed Monkey</x:String>
                    <x:String>Thomas's Langur</x:String>
                    <x:String>Purple-faced Langur</x:String>
                    <x:String>Gelada</x:String>
                </x:Array>
            </CollectionView.ItemsSource>
        </CollectionView>

```
4. Add the code `Forms.SetFlags("CollectionView_Experimental");` to the `AppDelegate.FinishedLaunching` before the `global::Xamarin.Forms.Forms.Init();` call.
5. Set the iOS app as the startup project and deploy

### Expected Behavior
The UI will contain the `Label` and the `CollectionView`.

### Actual Behavior
Only the `Label` shows.

### Notes
Either of the following will cause the `CollectionView` to show:
1. Setting the `VerticalOptions="FillAndExpand"` on the `CollectionView`
2. Wrapping the `MainPage` in a `NavigationPage`. 
3. Removing the `StackLayout` and the `Label` from the `MainPage.xaml`

### Basic Information

- Version with issue: 4.1.0.581479
- Last known good version: none.
- IDE: VS for Mac 8.1.5.9
- Platform Target Frameworks: 
  - iOS:  12.2 (Deployment Target set to 8.0, the default)
- Nuget Packages: Xamarin.Essentials 1.2.0

**Comment by rarelyleemallon** [comment]
Also occurred with CarouselView though the provided workaround from @jgold6 on removing the stacklayout or adding navigation page doesn't resolve. CarouselView is there as you can set the background colour but no date templates show.

**Comment by jgold6** [comment]
@rarelyleemallon 
That sounds like a different issue then. In this case not even the background of the CollectionView shows, where what you describe it seems that the CarouselView shows, but the items don't. Can you file a new issue for the issue you are describing?

**Comment by rarelyleemallon** [comment]
@jgold6 ah cool, yeah will get a test project together to show it

**Comment by hartez** [comment]
Confirmed. CollectionView shows up on Android and UWP; on iOS, the label takes up the whole screen.

[_6834 Repro.zip](https://github.com/xamarin/Xamarin.Forms/files/3387921/_6834.Repro.zip)

**Comment by samhouts** [comment]
closed by #6912
